src/main/java/adris/altoclef/tasks/misc/RepairToolTask.java

```diff
@@ -0,0 +1,197 @@
+import adris.altoclef.util.slots.PlayerSlot;
+import adris.altoclef.util.helpers.ItemHelper;
+import net.minecraft.item.Items;
+import baritone.api.utils.input.Input;
+import net.minecraft.entity.mob.ZombieEntity;
+import net.minecraft.entity.ExperienceOrbEntity;
+import adris.altoclef.util.csharpisbetter.TimerGame;
+import adris.altoclef.util.helpers.LookHelper;
+import baritone.api.utils.Rotation;
+import net.minecraft.entity.Entity;
+import net.minecraft.nbt.NbtCompound;
+import net.minecraft.nbt.NbtElement;
+import net.minecraft.item.ItemStack;
+import java.util.List;
+import java.util.Optional;
+import java.util.Arrays;
+
+public class RepairToolTask extends Task {
+
+    private ItemTarget[] _toRepair;
+    private boolean _finished;
+    private final TimerGame _throwTimer = new TimerGame(0.5);
+
+    public RepairToolTask(ItemTarget... toRepair) {
+        _toRepair = toRepair;
+    }
+    public RepairToolTask() { //If this task is called without itemtarget, repair anything we can
+        this(
+            new ItemTarget(ItemHelper.NETHERITE_ARMORS),
+            new ItemTarget(ItemHelper.NETHERITE_TOOLS),
+            new ItemTarget(Items.ELYTRA),
+            new ItemTarget(ItemHelper.DIAMOND_ARMORS),
+            new ItemTarget(ItemHelper.DIAMOND_TOOLS),
+            new ItemTarget(ItemHelper.IRON_ARMORS),
+            new ItemTarget(ItemHelper.IRON_TOOLS),
+            new ItemTarget(ItemHelper.GOLDEN_ARMORS),
+            new ItemTarget(ItemHelper.GOLDEN_TOOLS),
+            new ItemTarget(ItemHelper.STONE_TOOLS),
+            new ItemTarget(ItemHelper.LEATHER_ARMORS),
+            new ItemTarget(ItemHelper.WOODEN_TOOLS),
+            new ItemTarget(Items.FISHING_ROD),
+            new ItemTarget(Items.FLINT_AND_STEEL),
+            new ItemTarget(Items.CARROT_ON_A_STICK),
+            new ItemTarget(Items.SHEARS),
+            new ItemTarget(Items.BOW),
+            new ItemTarget(Items.SHIELD),
+            new ItemTarget(Items.TRIDENT),
+            new ItemTarget(Items.CROSSBOW),
+            new ItemTarget(Items.WARPED_FUNGUS_ON_A_STICK),
+            new ItemTarget(Items.BOW)
+        );
+    }   
+    @Override
+    protected void onStart(AltoClef mod) {
+        _throwTimer.reset();
+        _toRepair = Arrays.stream(_toRepair).filter(target -> needRepair(mod, target)).toArray(ItemTarget[]::new); //Filter out items type we can't repair
+        Optional<ItemTarget> ItemTargetOPTRepair = Arrays.stream(_toRepair).findFirst(); //Get the first element in our list
+        if (!ItemTargetOPTRepair.isPresent()) {
+            Debug.logWarning("There is nothing to repair");
+        }
+    }
+
+    @Override
+    protected Task onTick(AltoClef mod) {
+        //We start this task by filtering out every item type that we can't repair :
+        //All items without mending or with no damage
+        _toRepair = Arrays.stream(_toRepair).filter(target -> needRepair(mod, target)).toArray(ItemTarget[]::new);
```

Comment:
I'd make `_toRepair` `final` and make a local variable `ItemTarget[] shouldRepair` here or something. Instance variables should be final if they're compared to in `isEqual`, because otherwise the task may continuously re-run `onStart`